src/Reflection/Annotations/AnnotationMethodReflection.php

```diff
@@ -109,6 +110,13 @@ public function getThrowType(): ?Type
 
 	public function hasSideEffects(): TrinaryLogic
 	{
+		if (
```

Comment:
I took the logic from https://github.com/phpstan/phpstan-src/blob/97efd4d36b89fe8a5da39bc71abf5a25b44d4d26/src/Reflection/Php/PhpMethodReflection.php#L410-L415